Resources/Prototypes/Entities/Structures/Power/Generation/Tesla/energyball.yml

```diff
@@ -0,0 +1,132 @@
+- type: entity
+  id: BaseEnergyBall
+  abstract: true
+  components:
+  - type: Clickable
+  - type: Physics
+    bodyType: Dynamic
+    bodyStatus: InAir
+    sleepingAllowed: false
+  - type: CanMoveInAir
+  - type: ChasingWalk
+    minSpeed: 1
+    maxSpeed: 3
+    chasingComponent: 
+    - type: LightningTarget
+  - type: AmbientSound
+    volume: 3
+    range: 15
+    sound:
+      path: /Audio/Effects/tesla.ogg
+  - type: PointLight
+    enabled: true
+    radius: 5
+    color: "#4080FF"
+  - type: Appearance
+  - type: LightningArcShooter
+    arcDepth: 2
+    maxLightningArc: 1
+    shootMinInterval: 4
+    shootMaxInterval: 10
+    shootRange: 3
+    lightningPrototype: Lightning 
+
+- type: entity
+  id: TeslaEnergyBall
+  parent: BaseEnergyBall
+  name: ball lightning
+  description: A giant ball of pure energy. The space around it is humming and melting.
+  components: 
+  - type: Fixtures
+    fixtures:
+      EventHorizonCollider:
+        shape:
+          !type:PhysShapeCircle
+            radius: 0.55
+        hard: true
+        restitution: 0.8
+        density: 99999
+        mask:
+        - Opaque
+        layer:
+        - GlassLayer
+      EventHorizonConsumer:
+        shape:
+          !type:PhysShapeCircle
+            radius: 0.65
+        hard: false
+        mask:
+        - Opaque
+        layer:
+        - GlassLayer
+  - type: EventHorizon
```

Comment:
for some reason tesla started eating its micro lightning bolts again, and the lightning beams.